File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    
    clothes = [1 for i in range(n+2)]
    for i in lost:
        clothes[i] -= 1
    for i in reserve:
        clothes[i] += 1
    
    print(clothes)
    
    for i in range(1, len(clothes)):
        if clothes[i] == 0 and clothes[i-1] == 2:
            clothes[i] += 1
            clothes[i-1] -= 1
        elif clothes[i] == 0 and clothes[i+1] == 2:
            clothes[i] += 1
            clothes[i+1] -= 1
    
    for i in clothes:
        if i == 0:
            answer += 1
    answer = n - answer
    
    return answer

File: 프로그래머스/lv1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;

        int[] clothes = new int[n+2];
        
        for (int i = 0; i<n+2; i++){
            clothes[i] = 1;
        }
        
        for (int i = 0; i<reserve.length ; i++){
            clothes[reserve[i]] += 1;
        }
        
        for (int i = 0; i<lost.length ; i++){
            clothes[lost[i]] -= 1;
        }
        
        for (int i = 0; i< n+1; i++){
            if (clothes[i] == 0 && clothes[i-1] == 2){
                clothes[i] = 1;
                clothes[i-1] = 1;
            }
            else if(clothes[i] == 0 && clothes[i+1] == 2){
                clothes[i] = 1;
                clothes[i+1] = 1;
            }
        }
        for (int i = 1; i< n+1; i++){
            if (clothes[i] == 0){
                answer += 1;
            }
        }
        
        return n - answer;
        
    }
}